pkg/quota/admission/runonceduration/admission.go

```diff
@@ -0,0 +1,133 @@
+package runonceduration
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"reflect"
+	"strconv"
+
+	"k8s.io/kubernetes/pkg/admission"
+	kapi "k8s.io/kubernetes/pkg/api"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+
+	oadmission "github.com/openshift/origin/pkg/cmd/server/admission"
+	configlatest "github.com/openshift/origin/pkg/cmd/server/api/latest"
+	projectcache "github.com/openshift/origin/pkg/project/cache"
+	"github.com/openshift/origin/pkg/quota/admission/runonceduration/api"
+	"github.com/openshift/origin/pkg/quota/admission/runonceduration/api/validation"
+)
+
+func init() {
+	admission.RegisterPlugin("RunOnceDuration", func(client kclient.Interface, config io.Reader) (admission.Interface, error) {
+		pluginConfig, err := readConfig(config)
+		if err != nil {
+			return nil, err
+		}
+		return NewRunOnceDuration(pluginConfig), nil
+	})
+}
+
+func readConfig(reader io.Reader) (*api.RunOnceDurationConfig, error) {
+	config := &api.RunOnceDurationConfig{}
+	if reader == nil || reflect.ValueOf(reader).IsNil() {
+		return config, nil
+	}
+	configBytes, err := ioutil.ReadAll(reader)
+	if err != nil {
+		return nil, err
+	}
+	err = configlatest.ReadYAML(configBytes, config)
+	if err != nil {
+		return nil, err
+	}
+	errs := validation.ValidateRunOnceDurationConfig(config)
+	if len(errs) > 0 {
+		return nil, errs.ToAggregate()
+	}
+	return config, nil
+}
+
+func NewRunOnceDuration(config *api.RunOnceDurationConfig) admission.Interface {
+	return &runOnceDuration{
+		Handler: admission.NewHandler(admission.Create, admission.Update),
+		config:  config,
+	}
+}
+
+type runOnceDuration struct {
+	*admission.Handler
+	config *api.RunOnceDurationConfig
+	cache  *projectcache.ProjectCache
+}
+
+var _ = oadmission.WantsProjectCache(&runOnceDuration{})
+var _ = oadmission.Validator(&runOnceDuration{})
+
+func (a *runOnceDuration) Admit(attributes admission.Attributes) error {
+	if a.config == nil {
+		return nil
+	}
+	if a.config.ActiveDeadlineSecondsOverride == nil && !a.config.AllowProjectAnnotationOverride {
+		return nil
+	}
+	if attributes.GetResource() != kapi.Resource("pods") {
+		return nil
+	}
+	if len(attributes.GetSubresource()) > 0 {
```

Comment:
Again a matter of taste but I dislike seeing four if statements that all have the same outcome. This would look better to me:

    if a.config == nil  || 
         a.config.ActiveDeadlineSecondsOverride == nil && !a.config.AllowProjectAnnotationOverride ||
         attributes.GetResource() != kapi.Resource("pods") || attributes.GetSubresource() != "" {
                return nil
    }

Or even better:

    if a.config == nil  || !a.config.Enabled || a.config.ActiveDeadlineSecondsOverride == nil ||
         attributes.GetResource() != kapi.Resource("pods") || attributes.GetSubresource() != "" {
                return nil
    }

I leave it to your judgment though.